notfound/utils.py

```diff
@@ -25,14 +34,36 @@ def replace_uris(app, doctree, nodetype, nodeattr):
             if uri.startswith('../'):
                 uri = uri.replace('../', '')
 
+        imagedir = ''
+        if nodetype == docutils.nodes.image:
+            # Prefix the URL with ``Builder.imagedir`` to use the internal's
+            # Sphinx image handling if the node is an image
+            imagedir = '{imagedir}/'.format(
+                imagedir=app.builder.imagedir,
+            )
+
         if app.config.notfound_no_urls_prefix:
-            uri = '/{filename}'.format(
+            uri = '/{imagedir}{filename}'.format(
                 filename=uri,
+                imagedir=imagedir,
             )
         else:
-            uri = '/{language}/{version}/{filename}'.format(
+            uri = '/{language}/{version}/{imagedir}{filename}'.format(
```

Comment:
Feels like `imagedir` should just be part of `filename` rather than being a separate field.

Reply:
Yes. Although, since this prefix is automatically added by Sphinx (and the `filename` is what the user writes in the .rst file) I prefer to keep these things explicitly separated so it's easier to know what is what.